### ISSUE TITLE
error copying layers and metadata for container

### ISSUE THREAD

**Issue by manuelfrei** [containers/fuse-overlayfs#322]
Hi

We are using podman version 3.2.3 with fuse-overlayfs-1.6-1.module+el8.4.0+11822+6cc1e7d7.src.rpm. When we start ~20 rootless podman builds one or more builds fail with failures like the following:

error copying layers and metadata for container "85cd75cbdfefeca6ec6cd98aa53fca172db68dbf970f2141e9146fe6fda26f26": Error initializing source containers-storage:59854a5fed6358d7670c49745df86483137ab2832f28aaabb9546c8d7dbd18a3-working-container: error extracting layer "7fb95a7d17dc8334112aadd8cf7236722c5520374dfa4f3a902865d209a0a9a1": lgetxattr /opt/jenkins/.local/share/containers/storage/overlay/7ef7d32ec7d4b7a33fc14ceef832993e7f85a4f7b56dd4519311c187852abffc/merged/usr/share/zoneinfo/right/America/Noronha: no such file or directory

It looks like the fix https://github.com/containers/fuse-overlayfs/pull/300 . It should be backported to our release. Or is that a other issue?

podman info
host:
  arch: amd64
  buildahVersion: 1.21.3
  cgroupControllers:
  - cpuset
  - cpu
  - cpuacct
  - blkio
  - memory
  - devices
  - freezer
  - net_cls
  - perf_event
  - net_prio
  - hugetlb
  - pids
  - rdma
  cgroupManager: systemd
  cgroupVersion: v1
  conmon:
    package: conmon-2.0.29-1.module+el8.4.0+11822+6cc1e7d7.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.29, commit: ae467a0c8001179d4d0adf4ada381108a893d7ec'
  cpus: 128
  distribution:
    distribution: '"rhel"'
    version: "8.4"
  eventLogger: file
  hostname: adzh-srlp-ci28
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 4.18.0-305.17.1.el8_4.x86_64
  linkmode: dynamic
  memFree: 454478503936
  memTotal: 540479496192
  ociRuntime:
    name: runc
    package: runc-1.0.0-74.rc95.module+el8.4.0+11822+6cc1e7d7.x86_64
    path: /usr/bin/runc
    version: |-
      runc version spec: 1.0.2-dev
      go: go1.15.13
      libseccomp: 2.5.1
  os: linux
  remoteSocket:
    path: /run/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_NET_RAW,CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 0
  swapTotal: 0
  uptime: 25h 48m 24.47s (Approximately 1.04 days)
registries:
  search:
  - registry.access.redhat.com
  - registry.redhat.io
  - docker.io
store:
  configFile: /etc/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: fuse-overlayfs-1.6-1.module+el8.4.0+11822+6cc1e7d7.x86_64
      Version: |-
        fusermount3 version: 3.2.1
        fuse-overlayfs: version 1.6
        FUSE library version 3.2.1
        using FUSE kernel interface version 7.26
    overlay.mountopt: nodev,metacopy=on
  graphRoot: /var/lib/containers/storage
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 283
  runRoot: /run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 3.2.3
  Built: 1627370979
  BuiltTime: Tue Jul 27 09:29:39 2021
  GitCommit: ""
  GoVersion: go1.15.7
  OsArch: linux/amd64
  Version: 3.2.3

**Comment by manuelfrei** [comment]
Additional the Dockerfile for the podman build

`FROM jboss-eap-7/eap73-openjdk11-openshift-rhel8:7.3.8-1
COPY --chown=jboss:root content/common/ /
RUN mkdir /opt/batch-client/logs/ && chmod -R g+rw /opt/batch-client/logs/
ENTRYPOINT ["/docker-entrypoint.sh"]
CMD ["batch-client"]
COPY --chown=jboss:root content/specific/ /`

**Comment by giuseppe** [comment]
that looks like #300 could help here.

Would it be possible to try the last version from sources and see if it works better?

**Comment by manuelfrei** [comment]
Unfortunately, it does not work with the latest release 1.7.1 . I downloaded it and adjusted the path to this release in the storage.conf.

The same failures during build.
```
    overlay.mount_program:
      Executable: /opt/jenkins/fuse-overlayfs
      Package: Unknown
      Version: |-
        fusermount3 version: 3.2.1
        fuse-overlayfs: version 1.7.1
        FUSE library version 3.4.1
        using FUSE kernel interface version 7.27
```

**Comment by giuseppe** [comment]
do you have a reproducer?

**Comment by manuelfrei** [comment]
yes, the dockerfile is already above in the comments and i start the parallel builds with:

```
#!/bin/bash
x=1
while [ $x -le 20 ]
do
  echo "$x times"
  podman build -f Dockerfile --no-cache &> log$x.log &
  x=$(( $x + 1 ))
done
```

The content directory includes 100MB jar files

**Comment by manuelfrei** [comment]
I reduced the Dockerfile to:

```
FROM jboss-eap-7/eap73-openjdk11-openshift-rhel8:7.3.8-1
COPY --chown=jboss:root testc/ /
```

And created 100x 1MB Files in the directory testc

**Comment by giuseppe** [comment]
I am not able to reproduce locally, I've tried both on Fedora 34 and RHEL 8.4.

Could you strace the podman processes?

**Comment by manuelfrei** [comment]
I catched the strace.log and uploaded it to https://www.file.io/download/hTe0k2aqQhGU . Or are you interested in a special part of the strace?

I had to set the reproducer to parallel 30+ today for the error to occur. I don't know what else is affecting it.
We are using servers with 128 HT cores and NVME disks. When i start the script with 30 parallel builds it finished in 43 seconds.

**Comment by manuelfrei** [comment]
btw there are also failing builds when i start it with root

**Comment by manuelfrei** [comment]
As i understand when i start it with root fuse-overlayfs isn't included in the build process?

**Comment by giuseppe** [comment]
no, if you start it with root then there is no fuse-overlayfs involved.

Could you run with `strace -f`?  I am interested to see the `llistxattr` syscall failures

**Comment by manuelfrei** [comment]
I have added the strace -f to the script. The build now logically takes 15x longer. Unfortunately, I can't recreate the problem this way. Looks like there is a time component involved.

**Comment by manuelfrei** [comment]
I have a script run with errors and strace. And have provided the logfile of a podman build at: https://file.io/tmpYyVEy8dXN

**Comment by giuseppe** [comment]
I am not able to download that file, I've tried several times and I always get "504 Gateway Time-out".

Do you spot any `llistxattr` syscall failing with ENOENT?  Could you copy&paste here just the failing `llistxattr` calls?

**Comment by manuelfrei** [comment]
I greped over the file for llistxattr in combination with ENOENT and can't found any results. I uploaded the log again to: https://file.io/ORaQGll1FojD
This works only with "one" download.

**Comment by flouthoc** [comment]
@manuelfrei Please don't close this issue. We can move this to c/storage.
